File: VaccineDrugLocation/vaccine_drug_location_backend/vaccine_drug_location/src/main/java/dke/vaccine_location_drug/entity/Article.java
package dke.vaccine_location_drug.entity;

import jakarta.persistence.*;


@Entity
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long article_id;
    private String name;
    private int minAge;
    private int maxAge;
    private String type;

    // Getter und Setter

    public Long getId() {
        return article_id;
    }

    public void setId(Long id) {
        this.article_id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getMinAge() {
        return minAge;
    }

    public void setMinAge(int minAge) {
        this.minAge = minAge;
    }

    public int getMaxAge() {
        return maxAge;
    }

    public void setMaxAge(int maxAge) {
        this.maxAge = maxAge;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    // Weitere Methoden und Annotationen nach Bedarf
}

File: vaccine_drug_location_frontend/vaccine_drug_location/src/app/entities/Article.ts
export class Article {
  public id?: number;
  public name: string;
  public minAge: number;
  public maxAge: number;
  public type: string;
  public stock: number;

  constructor(
    id: number,
    name: string,
    minAge: number,
    maxAge: number,
    type: string,
    stock: number
  ) {
    this.id = id;
    this.name = name;
    this.minAge = minAge;
    this.maxAge = maxAge;
    this.type = type;
    this.stock = stock;
  }
}
